cmd/kubeadm/app/phases/upgrade/postupgrade_test.go

```diff
@@ -163,14 +147,9 @@ func TestWriteKubeletConfigFiles(t *testing.T) {
 		},
 	}
 	for _, tc := range testCases {
-		err := WriteKubeletConfigFiles(tc.cfg, tc.patchesDir, tc.dryrun, os.Stdout)
-		if err != nil && tc.errPattern != "" {
-			if match, _ := regexp.MatchString(tc.errPattern, err.Error()); !match {
-				t.Fatalf("Expected error contains %q, got %v", tc.errPattern, err.Error())
-			}
-		}
-		if err == nil && len(tc.errPattern) != 0 {
-			t.Fatalf("WriteKubeletConfigFiles didn't return error expected %s", tc.errPattern)
+		err := WriteKubeletConfigFiles(tc.cfg, t.TempDir(), tc.patchesDir, true, os.Stdout)
```

Comment:
this could have been only one t.TempDir() on the top of the function, but this is fine too.
/lgtm

Reply:
/lgtm cancel

actually this is not using subtests, so either put it once on top of the function or add a subtest.

Reply:
I will better add one on top and use it across.